hw/ip/spi_device/lint/spi_device.waiver

```diff
@@ -121,6 +121,9 @@ waive -rules RESET_MUX -location {spi_device.sv} \
 waive -rules RESET_MUX -location {spid_status.sv} \
       -regexp {Asynchronous reset 'status_fifo_rst_n' is driven by a multiplexer here, used as a reset 'rst_wr_ni'} \
       -comment "Scan reset mux, but need to have asynchronous reset"
+waive -rules RESET_USE -location {spid_dpram.sv} \
+      -regexp {'(rst_sys_ni|rst_spi_ni)' is connected to 'prim_ram_1r1w_async_adv' port 'rst_a_ni', and used as an asynchronous reset or set} \
+      -comment "Asynchronous reset use in 1r1w ram primitive is legal."
```

Comment:
I was suspicious because we don't need to waive similar warnings for the 2p variant in Earlgrey and I think the underlying issue here is that inside `prim_generic_ram_1r1w` which is instantiated inside `prim_ram_1r1w_async_adv`, the reset inputs are tied off:
```sv
  logic unused_signals;
  assign unused_signals = ^{cfg_i, rst_a_ni, rst_b_ni};
```
So I think this waiver here makes sense and is fine. I am leaving the comment for others.

We may want to make the comment more precise but I don't consider this a must.